blockchain/reactor.go

```diff
@@ -267,6 +274,22 @@ func (bcR *BlockchainReactor) SetEventSwitch(evsw *events.EventSwitch) {
 	bcR.evsw = evsw
 }
 
+type proxyAppWrapper struct {
+	app   proxy.AppContext
+	ready chan struct{}
+	done  chan struct{}
+}
+
+func (bcR *BlockchainReactor) ResetProxyApp(proxyApp proxy.AppContext, ready, done chan struct{}) {
```

Comment:
might be better to just add a proxyAppMtx and use that instead of fitting it into the poolRoutine select loop